sql/mysqld.cc

```diff
@@ -2057,6 +2057,7 @@ class Set_wsrep_kill_conn : public Do_THD_Impl {
   int get_dump_thread_count() const { return m_dump_thread_count; }
 
   virtual void operator()(THD *killing_thd) {
+    if (killing_thd == current_thd) return;
```

Comment:
Was it OK before without this check added in existing places where `Set_wsrep_kill_conn` was instantiated? I mean in everywhere else rather than in `wsrep_close_client_connections()` that you added in this commit?

Reply:
Set_wsrep_kill_conn is currently not used anywhere in the code.
If we look to the comments on `wsrep_close_client_connections ()` https://github.com/percona/percona-xtradb-cluster/blob/8.0/sql/mysqld.cc#L9393 - it was meant to mimic the behavior of `close_connections()` https://github.com/percona/percona-xtradb-cluster/blob/8.0/sql/mysqld.cc#L2232 - which first sends a kill, and later tries to close it abruptly.

`if (killing_thd == current_thd) return;` ensures that the thread executing `SET GLOBAL wsrep_cluster_address = ''` or `SET GLOBAL wsrep_reject_queries=ALL_KILL` will not be also killed. 

However, I've re-checked what Set_wsrep_kill_conn and Call_wsrep_close_client_conn will signal/close and they are not doing the same thing. I will create a new method to signal only client `_client_` part from Call_wsrep_close_client_conn and will leave Set_wsrep_kill_conn intact.

Reply:
> or SET GLOBAL wsrep_reject_queries=ALL_KILL will not be also killed.

From https://galeracluster.com/library/documentation/mysql-wsrep-options.html#wsrep-reject-queries:
ALL_KILL The node enables this feature. It rejects all queries and kills existing client connections without waiting, including the current connection.

Reply:
I would say this is one of the differences between pxc and galera. Currently we don't kill the thd calling the SET:

https://www.percona.com/doc/percona-xtradb-cluster/5.7/wsrep-system-index.html#wsrep_reject_queries. 

New kill function and Call_wsrep_close_client_conn (currently used) have the same criteria about which threads to kill/close.